partiql-parser/src/main/kotlin/org/partiql/parser/internal/PartiQLParserDefault.kt

```diff
@@ -2202,41 +2192,6 @@ internal class PartiQLParserDefault : PartiQLParser {
             else -> throw error(ctx, "Expected set quantifier ALL or DISTINCT")
         }
 
-        /**
-         * With the <string> and <int> nodes of a literal time expression, returns the parsed string and precision.
-         * TIME (<int>)? (WITH TIME ZONE)? <string>
-         */
-        private fun getTimeStringAndPrecision(
-            stringNode: TerminalNode,
-            integerNode: TerminalNode?,
-        ): Pair<String, Int> {
-            val timeString = stringNode.getStringValue()
-            val precision = when (integerNode) {
-                null -> {
-                    try {
-                        getPrecisionFromTimeString(timeString)
-                    } catch (e: Exception) {
-                        throw error(stringNode.symbol, "Unable to parse precision.", e)
-                    }
-                }
-                else -> {
-                    val p = integerNode.text.toBigInteger().toInt()
-                    if (p < 0 || 9 < p) throw error(integerNode.symbol, "Precision out of bounds")
-                    p
-                }
-            }
-            return timeString to precision
-        }
-
-        private fun getPrecisionFromTimeString(timeString: String): Int {
-            val matcher = GENERIC_TIME_REGEX.toPattern().matcher(timeString)
-            if (!matcher.find()) {
-                throw IllegalArgumentException("Time string does not match the format 'HH:MM:SS[.ddd....][+|-HH:MM]'")
-            }
-            val fraction = matcher.group(1)?.removePrefix(".")
-            return fraction?.length ?: 0
-        }
```

Comment:
(self-review): this prior parsing logic was deleted for the following reasons
- was eagerly inferring the seconds precision for TIME and TIMESTAMP when the precision argument was unspecified. Thus it was classifying `TIME '12:34:56'` as the same as `TIME (0) '12:34:56'"`
- applying the time string's regex check for timestamp
- only performing the string regex check when the precision was unspecified